Framework/API/src/InfoComponentVisitor.cpp

```diff
@@ -126,6 +134,18 @@ InfoComponentVisitor::assemblySortedDetectorIndices() const {
 const std::vector<Mantid::Geometry::ComponentID> &
 InfoComponentVisitor::componentIds() const {
   return m_componentIds;
+  /*
+   * TODO. There is an issue here that will need to be addressed.
+   * Detectors can be dropped (see above). This is a workaround of Instrument
+   * 1.0 for
+   * the fact that in some cases we have IDFs that provide an invalid ID for a
+   * detector.
+   * However, we use a fixed offset for the non-detector component indexes
+   * calculated to be the
+   * exact size of the number of detectors we expect to get. When we drop
+   * detectors we introduce gaps in
+   * an otherwise contiguous range.
+  */
```

Comment:
Correct. This is a note for the future. Doesn't currently cause an issue. I think the correct fix should be at the IDF level, I don't like this dropped detector thing. Failing that being possible, we would have to "shift down" or index ranges before actually returning the containers we create.